### PR TITLE
[WIP] Cooper client  in react guide

### PR DESCRIPTION
**Pull request by oliverochman**

### CODE REVIEW

File: cooper_client_react.md [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+          if (distance >= min && distance <= max) {
+              ratingIndex = index;
+          }
+      }
+  });
+  return ratings[ratingIndex];
+}
+```
+
+If you run the test now, both tests should go green. Ok s lets run the feature tests agin now, everything should go green right. We have a component that displays the correct result.
```
Comment (magnus-thor) [inline cooper_client_react.md]: Extra `s` in there `Ok s lets`, and there is a typo in again `tests agin now`

File: cooper_client_react.md [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+
+BUT, as you can see when we run the feature tests, nothing has changed. Thats because we havent actually rendered the `DiplayCooperResult` component in the `App` component. We havent passed in any props to the `DisplayCooperResult` component. We havent even saved what gets written in the input fields. So the actual `DiplayCooperResult` component works, but not with rest of the application. This is the reason they are called component tests.
```
Comment (magnus-thor) [inline cooper_client_react.md]: `havent` should be `haven't`

File: cooper_client_react.md [hunk 3]
```diff
@@ -0,0 +1,383 @@
...
+    verbose: true,
+    preset: "jest-puppeteer",
+    testRegex: ".feature\\.js$"
+}; 
+```
+
+```js
+module.exports = {
+  launch: {
+      headless: false,
```
Comment (magnus-thor) [inline cooper_client_react.md]: In the bmi guide you use `headless: process.env.CI === 'true',` any reason why you dont use it here?

File: cooper_client_react.md [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+  "start": "react-scripts start",
+  "build": "react-scripts build",
+  "test": "react-scripts test",
+  "eject": "react-scripts eject",
+  "test:features": "jest -c jest.config.js"
+}
+```
+It is time to add our first test now. Run:
+
+`mkdir src/__features__`
```
Comment (magnus-thor) [inline cooper_client_react.md]: missing `$` in front of the command

File: cooper_client_react.md [hunk 5]
```diff
@@ -0,0 +1,383 @@
...
+
+`touch src/__features__/CooperCalculation.feature.js`
```
Comment (magnus-thor) [inline cooper_client_react.md]: same

File: cooper_client_react.md (hunk 3 above)
Comment (oliverochman) [reply cooper_client_react.md]: We have been doing so many changes to this file over the time so I cant really answer the exact reason why we dont use it here. But I will update this file with the latest solution that seems to work well.

File: cooper_client_react.md (hunk 4 above)
Comment (oliverochman) [reply cooper_client_react.md]: fixed

File: cooper_client_react.md (hunk 5 above)
Comment (oliverochman) [reply cooper_client_react.md]: fixed

File: cooper_client_react.md (hunk 1 above)
Comment (oliverochman) [reply cooper_client_react.md]: fixed

File: cooper_client_react.md (hunk 2 above)
Comment (oliverochman) [reply cooper_client_react.md]: fixed

**Pull request closed without merging**